Automattic/wp-calypso
# Supports checking only initial fragment in url path.

#### Changes proposed in this Pull Request

* Uses a new `itemLinkMatches` function specific for nav unification
* Matches the links by comparing only the first fragment of the path eg `posts`
* Includes a workaround for jetpack types which all start with `types` but differ on the second fragment eg `jetpack-portfolio`
* Includes a temporary fix for sharing buttons being under both tools and settings.

#### Testing instructions



* Navigate around to calypso
* Especially check screens which include secondary navigation eg `pages`, `portfolio`, `tools`
* Make sure that at any case the correct menu is highlighted

**Automated** 
`yarn run test-client client/my-sites/sidebar-unified/test/index.js`

Before | After
-------|------
![](https://cln.sh/iXvRkn+) | ![](https://cln.sh/hUObUm+)



Fixes #49310
